tt_metal/impl/debug/watcher_server.cpp

```diff
@@ -899,17 +888,108 @@ void watcher_init(Device *device) {
     else
         watcher_enable_init_val.push_back(WatcherDisabled);
 
-    // Initialize worker cores debug values
     CoreCoord grid_size = device->logical_grid_size();
+
+    // Initialize Debug Delay feature
+    std::map<CoreCoord, debug_insert_delays_msg_t> debug_delays_val;
+    for (tt::llrt::RunTimeDebugFeatures delay_feature = tt::llrt::RunTimeDebugFeatureReadDebugDelay;
+         (int)delay_feature <= tt::llrt::RunTimeDebugFeatureAtomicDebugDelay;
+         delay_feature = (tt::llrt::RunTimeDebugFeatures)((int)delay_feature + 1)) {
+        vector<chip_id_t> chip_ids = tt::llrt::OptionsG.get_feature_chip_ids(delay_feature);
+        bool this_chip_enabled = tt::llrt::OptionsG.get_feature_all_chips(delay_feature) ||
+                                 std::find(chip_ids.begin(), chip_ids.end(), device->id()) != chip_ids.end();
+        if (this_chip_enabled) {
+            static_assert(sizeof(debug_sanitize_noc_addr_msg_t) % sizeof(uint32_t) == 0);
+            debug_insert_delays_msg_t delay_setup;
+
+            // Create the mask based on the feature
+            uint32_t hart_mask = tt::llrt::OptionsG.get_feature_riscv_mask(delay_feature);
+            switch (delay_feature) {
+                case tt::llrt::RunTimeDebugFeatureReadDebugDelay: delay_setup.read_delay_riscv_mask = hart_mask; break;
+                case tt::llrt::RunTimeDebugFeatureWriteDebugDelay:
+                    delay_setup.write_delay_riscv_mask = hart_mask;
+                    break;
+                case tt::llrt::RunTimeDebugFeatureAtomicDebugDelay:
+                    delay_setup.atomic_delay_riscv_mask = hart_mask;
+                    break;
+                default: break;
+            }
+
+            for (CoreType core_type : {CoreType::WORKER, CoreType::ETH}) {
+                vector<CoreCoord> delayed_cores = tt::llrt::OptionsG.get_feature_cores(delay_feature)[core_type];
+                for (tt_xy_pair logical_core : delayed_cores) {
+                    CoreCoord phys_core;
+                    bool valid_logical_core = true;
+                    try {
+                        phys_core = device->physical_core_from_logical_core(logical_core, core_type);
+                    } catch (std::runtime_error &error) {
```

Comment:
Should we print a message saying that the given core doesn't map to anything and will be ignored? We do that for dprint